### ISSUE TITLE
Model Training, Optimization, and Validation

### ISSUE THREAD

**Issue by Fariba-Kamani** [Fariba-Kamani/Heritage-House-Price-Insight-Predictor#10]
As a **data scientist**, I want to **optimize model hyperparameters using cross-validation**, so that I can **maximize prediction accuracy**.

**Acceptance Criteria:**

1- Hyperparameter optimization is performed.

2- R2 ≥ 0.75 is achieved on both train and test sets.

**Comment by Fariba-Kamani** [comment]
Must have
